source/develop/release-management/features/storage/4k-blocks.md

```diff
@@ -0,0 +1,93 @@
+---
+title: 4k block device support for the storage domains
+category: feature
+authors: dchaplyg
+feature_name: 4k block device support for the storage domains
+feature_modules: vdsm
+feature_status: Planning
+---
+
+# 4k block device support for the storage domains
+
+# Summary
+
+VDSM assumes, that 512b is a minimally accessible block on the storage device, which can be written or read atomically. That was true for a long time, but now most of the storage devices changed their block size to the 4096 bytes (4 kilobytes). It is still compatible with VDSM, case while we are requesting 512 atomic read or write, underlying storage will atomically read 4096 bytes, throw away 3584 bytes and return just 512. As it still acts same way, it has no impact on the VDSM. 
+
+Unfortunately, some 4k block storage devices either refuse to read/write smaller blocks atomically (some SAN appliances) or have a performance impact while emulating 512b block device (like VDO). 
+
+Therefore without 4k block device support oVirt either can't use those devices as a storage domain or will have performance degradation on those domains.
+
+## Owner
+
+*   Name: [Denis Chaplygin](https://github.com/akashihi)
+*   Email: <dchaplyg@redhat.com>
+
+## Current status
+
+*   Target Release: 4.3
+*   Status: Planning
+*   Last updated: July 31, 2018
+
+# Implementation details
+
+VDSM heavily relies on 512 block support and it is hardcoded to a lots of places.
+
+The general approach is to refactor all block size dependent code to use block size from the storage domain, while keeping it hardcoded to 512b in it and in a second step, implement block size detection at the SD level and replace hardcoded value with actual value.
+
+Subsystem specific notes are below:
+
+## Sanlock
+
+Sanlock is used by oVirt for locking and fencing shared resources. Latest released sanlock have native support for 4k devices, but, unfortunately, at the moment it work only with block devices, not with filesystems. Speaking precisely, sanlock on filesystems always assumes, that block size of the device is 512b. With patch [1] it will be possible, to specify desired block size, but it is still not able to detect block size of the filesystem backing device.
+
```

Comment:
Is the patch merged? What's the impact of not being able to auto-detect?

Reply:
It is merged, but not yet released.

Reply:
It does not expose the capability via the python binding yet.